dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/plugins/OidcBaseClientValidator.java

```diff
@@ -391,12 +396,15 @@ protected void validateAppType() throws OidcServerException {
         if (!OidcOAuth20Util.isNullEmpty(appType) &&
                 !appType.equals(OIDCConstants.OIDC_CLIENTREG_PARM_NATIVE) &&
                 !appType.equals(OIDCConstants.OIDC_CLIENTREG_PARM_WEB)) {
-            String description = TraceNLS.getFormattedMessage(OidcBaseClientValidator.class,
-                    MESSAGE_BUNDLE,
-                    "OAUTH_CLIENT_REGISTRATION_VALUE_NOT_SUPPORTED",
-                    new Object[] { appType, OIDCConstants.OIDC_CLIENTREG_APP_TYPE },
-                    "CWWKS1442E: The value {0} is not a supported value for the {1} client registration metadata field.");
-            throw new OidcServerException(description, OIDCConstants.ERROR_INVALID_CLIENT_METADATA, HttpServletResponse.SC_BAD_REQUEST);
+
+            BrowserAndServerLogMessage errorMsg = new BrowserAndServerLogMessage(tc, locale, "OAUTH_CLIENT_REGISTRATION_VALUE_NOT_SUPPORTED", new Object[] { appType, OIDCConstants.OIDC_CLIENTREG_APP_TYPE });
+            throw new OidcServerException(errorMsg, OIDCConstants.ERROR_INVALID_CLIENT_METADATA, HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
This looks like what we're trying to achieve 👍 . The only thing is that we might be able to remove the `locale` argument from the `BrowserAndServerLogMessage` constructor since we're actually looking up the appropriate locale elsewhere in the code (in `OAuth20EndpointServices` when we catch this exception).